index.js

```diff
@@ -106,6 +118,11 @@
     return f(this.value);
   };
 
+  //  Just#concat :: m a -> m a
+  Just.prototype.concat = function(maybe) {
+    return maybe instanceof Just ? Just(this.value.concat(maybe.value)) : this;
+  };
+
```

Comment:
> I wasn't expecting it to solve all my problems, mostly to reduce null-checking. Javascript code that's full of null-checks can be more nicely written with the help of some judiciously-placed Maybes.

Sorry, I think my remark was a little snarky. That wasn't my intention.

You're making perfect sense. My qualm is just that the other functions don't do any checking, but you want `concat` to do a bunch. I don't understand why `concat` needs to be special. `map` doesn't ensure that its argument is a function, `ap` doesn't ensure that the `Just` holds a function or that its argument is a `Maybe a`, `chain` doesn't ensure that the argument is a function. And none of them check to ensure that the return value is the proper type. Why should `concat` check that it holds a value that is a `Semigroup a`, and that its argument is a `Maybe` holding the same `Semigroup a`?